src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.AbstractRemoveDocumentUndoUnit.cs

```diff
@@ -30,12 +35,22 @@ public override void Do(IOleUndoManager pUndoManager)
                 var currentSolution = Workspace.CurrentSolution;
                 var fromProject = currentSolution.GetProject(FromProjectId);
 
-                if (fromProject != null &&
-                    GetDocumentIds(fromProject).Contains(DocumentId))
+                if (fromProject == null)
+                {
+                    return;
+                }
+
+                if (GetDocumentIds(fromProject).Contains(DocumentId))
                 {
                     var updatedProject = fromProject.RemoveDocument(DocumentId);
                     Workspace.TryApplyChanges(updatedProject.Solution);
                 }
+
+                if (CreatedFolder.Any())
+                {
+                    var project = Workspace.TryGetDTEProject(FromProjectId);
+                    project?.FindFolder(CreatedFolder)?.Remove();
+                }
```

Comment:
needs comment as to why this isn't done in Workspace.TryApplyChanges.  i think i know hte answer, but we need to doc.

Reply:
I reverted the change to this.